pkgs/development/compilers/purescript/purescript/generate.sh

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p cabal2nix
+
+set -x
+set -eo pipefail
+
+cabal2nix cabal://bower-json-1.0.0.1 > generated/bower-json.nix
+cabal2nix cabal://language-javascript-0.7.0.0 > generated/language-javascript.nix
+cabal2nix https://hackage.haskell.org/package/lens-4.19.2/revision/6.cabal > generated/lens.nix
+cabal2nix cabal://memory-0.15.0 > generated/memory.nix
+cabal2nix cabal://semialign-1.1.0.1 > generated/semialign.nix
+cabal2nix cabal://vector-0.12.3.1 > generated/vector.nix
+cabal2nix cabal://witherable-0.4.1 > generated/witherable.nix
+cabal2nix --revision v0.15.2 https://github.com/purescript/purescript > generated/purescript.nix
```

Comment:
It'd be nice if we could generate this file automatically as well.

One question I had though about this.  Are you sure that the versions of the Haskell packages in the package set produced with these overrides is identical to the package set you get when building with `stack` in the PureScript repo?  Are there no transitive dependencies that also need to be overridden?  How did you confirm this?  Do you have some script or something?  Or did you do it by hand?

If you did it by hand, it seems like it would be quite annoying to have to do it every time there is a new version of PureScript released.

Reply:
> Are you sure that the versions of the Haskell packages in the package set produced with these overrides is identical to the package set you get when building with `stack` in the PureScript repo?

On the contrary, I am quite sure that they are *not* identical.  That list was produced, and minimized, by hand.

> If you did it by hand, it seems like it would be quite annoying to have to do it every time there is a new version of PureScript released.

Yes, although my bigger concern is that this expression will become broken due to changes in `release-haskell.nix`, since I tried to maximize the number of packages taken from there (and minimize the number for which `generate.sh` runs `cabal2nix`).

This is what brought me to write [this comment](https://github.com/NixOS/nixpkgs/pull/176998#issuecomment-1150777650).  At the moment that is the only strategy I can think of that would keep this expression robust and maintainable.